docs/package.json

```diff
@@ -109,26 +110,39 @@
     "react-runner": "^1.0.3",
     "react-simple-code-editor": "^0.13.1",
     "react-swipeable-views": "^0.14.0",
+    "react-swipeable-views-utils": "^0.14.0",
     "react-transition-group": "^4.4.5",
     "react-virtuoso": "^4.5.0",
     "react-window": "^1.8.9",
     "recharts": "2.7.3",
     "rimraf": "^5.0.1",
     "styled-components": "^5.3.11",
-    "stylis": "^4.2.0",
+    "stylis": "4.2.0",
     "stylis-plugin-rtl": "^2.1.1",
     "stylis-plugin-rtl-sc": "npm:stylis-plugin-rtl@^1.1.0",
     "use-count-up": "^3.0.1",
-    "webfontloader": "^1.6.28",
     "webpack-bundle-analyzer": "^4.9.1"
   },
   "devDependencies": {
     "@babel/plugin-transform-react-constant-elements": "^7.22.5",
     "@babel/preset-typescript": "^7.22.5",
-    "@types/marked": "^5.0.1",
-    "@types/recharts": "^2.0.0",
+    "@mui-internal/docs-utilities": "^1.0.0",
+    "@types/chai": "^4.3.5",
+    "@types/node": "^18.17.6",
+    "@types/prop-types": "^15.7.5",
+    "@types/react": "^18.2.21",
+    "@types/react-dom": "18.2.7",
```

Comment:
`yarn docs:dev` now returns a new warning:

<img width="1391" alt="Screenshot 2023-09-11 at 02 21 26" src="https://github.com/mui/material-ui/assets/3165635/30be15bf-3b59-408a-866a-2b3f261ddbfd">

How about we move all the `@types` dependencies of docs to be devDependencies?

Reply:
> I see a lot more installation warnings (yarn)

We've got more dependencies now, so it's expected that the same errors will appear in more places, where the dependencies are defined.
However, some of them look like false positives (react as peer dependency). I'll see what I can do about them.

> why is @types/react not set only once in the root package.json workspace

Because it's used by individual packages, not just the scripts at the workspace root.

> How about we move all the @types dependencies of docs to be devDependencies?

Yup, types should be in dev dependencies. I'll fix it.

Reply:
> However, some of them look like false positives (react as peer dependency). I'll see what I can do about them.

👍 Many of them are top-level packages that define react as a peer dependency, but we don't have react as a workspace root dependency, so there is no strong guarantee about which version of react ends up adjacent to them in the node_modules folder. Just installing `react` and `react-dom` as a dev dependency in the root should solve a lot of these false positives.

The others that complain about incompatible react version are more problematic. These may work now, but as the authors don't guarantee compatibility with React 18, technically there may be breaking changes introduced for these packages on minor React versions

Reply:
Yes, pnpm requires packages to be defined where they are used.

Reply:
> Yes, pnpm requires packages to be defined where they are used.

pnpm doesn't hoist all packages to a top level node_modules folder as yarn does. Instead it creates node_modules folders with only the packages that are defined in package.json. As such it can't access transitive dependencies or dependencies that are defined by other workspaces. But dependency resolution at runtime is still just done through the node resolution algorithm, and that just traverses upwards until it finds a node_modules folder with the package in. it should thus be able to just access packages from a workspace that are dependencies of the workspace root. Defining `@types/react` at workspace root only should just work.

Reply:
> Defining @types/react at workspace root only should just work. (as far as I understand)

I love the potential of this change (forcing a default of sharing the same version while keeping flexibility).

Reply:
Yes, this should work, but why would you want to treat @types/react differently than the rest of the dependencies? Versions are handled by renovate anyway, so updating in individual packages is as easy as in workspace root.

Reply:
For the record, my strictly personal preference would also be to be explicit in defining dependencies. It makes the workspaces more self-contained. I don't think there is much practical benefit of one method over the other so either way is fine for me.

> forcing a default of sharing the same version while keeping flexibility

For reference, `pnpm` has an [RFC](https://github.com/gluxon/pnpm-rfcs/blob/d93a38b561731240bc9b36c69cd646429e906cf1/text/0001-catalogs.md) that should help with enforcing a single version across all workspaces. There also seems to be an [intent to implement](https://github.com/pnpm/pnpm/issues/7072).

Reply:
> I agree, my strictly personal preference would also be to be explicit in declaring all dependencies per workspace. It makes the workspaces more self-contained.

What's the value of having the dev dependencies that are used in 80% of the codebase to be defined in each workspace (where used)?

I'm blind to the value (the closest I can understand would be allowing us to move from yarn to pnpm to have faster npm install). Otherwise, I don't understand the rationale. I feel for the problems it creates, e.g. https://github.com/mui/material-ui/pull/38920#discussion_r1322304587, the problem [the RFC motivation](https://github.com/gluxon/pnpm-rfcs/blob/d93a38b561731240bc9b36c69cd646429e906cf1/text/0001-catalogs.md#motivation).

Reply:
> I'm blind to the value

That's why I call it a strictly personal preference. I enjoy the idea that when I can copy a workspace folder to my desktop and run `yarn` inside of it and everything just works it's a sign of true encapsulation. Meaning the workspace function is to link packages together and nothing else and all other code sharing between packages happens through explicit dependencies.

But truely personal preference in my opinion, the monorepo has much more important issues than this one.

Reply:
> seeing this RFC reminds me that it probably makes sense to take pnpm-workspaces.yaml into account for calculating the pnpm cache key in CI.

@Janpot, for now, I followed the official guidelines about setting up the cache. When the catalogs are introduced, we'll have to adjust our setup anyway, so then we can add the workspace definition to the cache key.